Multiple possible bean candidates found for PubSubMessageReceiverExceptionHandler

I'm following [Micronaut GCP guide](https://micronaut-projects.github.io/micronaut-gcp/latest/guide/#errorHandling) and bumped into this problem:

> `NonUniqueBeanException: Multiple possible bean candidates found: [io.micronaut.gcp.pubsub.exception.DefaultPubSubMessageReceiverExceptionHandler, com.example.MyErrorHandler]`


### Steps to Reproduce

1. `mn create-app --features gcp-pubsub com.example.foobar`
2. Create class `MyErrorHandler`:
```java
@Primary @Singleton
public class MyErrorHandler implements PubSubMessageReceiverExceptionHandler {
    @Override
    public void handle(PubSubMessageReceiverException exception) { 
        /* ... */
    }
}
```
3. `./gradlew run`


### Expected Behaviour

I was expecting that annotating my exception handler with `@Primary` would suffice to override the default one provided by Micronaut, as explained in the documentation:

> You define a new implementation of [PubSubMessageReceiverExceptionHandler](https://micronaut-projects.github.io/micronaut-gcp/latest/api/io/micronaut/gcp/pubsub/exception/PubSubMessageReceiverExceptionHandler.html) and annotate it as `@Primary` overriding the default one.


### Actual Behaviour

Stacktrace:

```
Message: Multiple possible bean candidates found: [io.micronaut.gcp.pubsub.exception.DefaultPubSubMessageReceiverExceptionHandler, com.example.MyErrorHandler]
Path Taken: new PubSubConsumerAdvice(BeanContext beanContext,ConversionService conversionService,PubSubMessageSerDesRegistry serDesRegistry,SubscriberFactory subscriberFactory,GoogleCloudConfiguration googleCloudConfiguration,PubSubConfigurationProperties pubSubConfigurationProperties,PubSubBinderRegistry binderRegistry,[PubSubMessageReceiverExceptionHandler exceptionHandler])
io.micronaut.context.exceptions.DependencyInjectionException: Failed to inject value for parameter [exceptionHandler] of class: io.micronaut.gcp.pubsub.intercept.PubSubConsumerAdvice

Message: Multiple possible bean candidates found: [io.micronaut.gcp.pubsub.exception.DefaultPubSubMessageReceiverExceptionHandler, com.example.MyErrorHandler]
Path Taken: new PubSubConsumerAdvice(BeanContext beanContext,ConversionService conversionService,PubSubMessageSerDesRegistry serDesRegistry,SubscriberFactory subscriberFactory,GoogleCloudConfiguration googleCloudConfiguration,PubSubConfigurationProperties pubSubConfigurationProperties,PubSubBinderRegistry binderRegistry,[PubSubMessageReceiverExceptionHandler exceptionHandler])
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:1566)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConstructorArgument(AbstractInitializableBeanDefinition.java:1041)
	at io.micronaut.gcp.pubsub.intercept.$PubSubConsumerAdvice$Definition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2232)
	at io.micronaut.context.DefaultBeanContext.addCandidateToList(DefaultBeanContext.java:3595)
	at io.micronaut.context.DefaultBeanContext.getBeanRegistrations(DefaultBeanContext.java:3471)
	at io.micronaut.context.DefaultBeanContext.streamOfType(DefaultBeanContext.java:906)
	at io.micronaut.context.DefaultBeanContext.streamOfType(DefaultBeanContext.java:891)
	at io.micronaut.context.DefaultBeanContext.streamOfType(DefaultBeanContext.java:873)
	at io.micronaut.context.DefaultBeanContext.lambda$initializeContext$31(DefaultBeanContext.java:1881)
	at java.base/java.util.HashMap.forEach(HashMap.java:1336)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1880)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:235)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3356)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:242)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:181)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:71)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:311)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:297)
	at com.example.Application.main(Application.java:25)
Caused by: io.micronaut.context.exceptions.NonUniqueBeanException: Multiple possible bean candidates found: [io.micronaut.gcp.pubsub.exception.DefaultPubSubMessageReceiverExceptionHandler, com.example.MyErrorHandler]
	at io.micronaut.context.DefaultBeanContext.findConcreteCandidate(DefaultBeanContext.java:2325)
	at io.micronaut.context.DefaultApplicationContext.findConcreteCandidate(DefaultApplicationContext.java:451)
	at io.micronaut.context.DefaultBeanContext.lastChanceResolve(DefaultBeanContext.java:3159)
	at io.micronaut.context.DefaultBeanContext.findConcreteCandidateNoCache(DefaultBeanContext.java:3046)
	at io.micronaut.context.DefaultBeanContext.findConcreteCandidate(DefaultBeanContext.java:2960)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2646)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1554)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:1544)
	... 19 common frames omitted
Caused by: io.micronaut.context.exceptions.NonUniqueBeanException: Multiple possible bean candidates found: [io.micronaut.gcp.pubsub.exception.DefaultPubSubMessageReceiverExceptionHandler, com.example.MyErrorHandler]

```


### Workaround

Instead of:

```java
@Primary
```

Use:

```java
@Replaces(bean = DefaultPubSubMessageReceiverExceptionHandler.class)
```


### Environment Information

- **Operating System**: Windows 10
- **Micronaut Version:** 3.0.0-M5
- **JDK Version:** openjdk version "11.0.11" 2021-04-20

--- a/gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/exception/DefaultPubSubMessageReceiverExceptionHandler.java
+++ b/gcp-pubsub/src/main/java/io/micronaut/gcp/pubsub/exception/DefaultPubSubMessageReceiverExceptionHandler.java
@@ -15,7 +15,6 @@
  */
 package io.micronaut.gcp.pubsub.exception;
 
-import io.micronaut.context.annotation.Primary;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
@@ -31,7 +30,6 @@ import jakarta.inject.Singleton;
  * @since 2.0.0
  */
 @Singleton
-@Primary
 public class DefaultPubSubMessageReceiverExceptionHandler implements PubSubMessageReceiverExceptionHandler {
 
     private final Logger logger = LoggerFactory.getLogger(DefaultPubSubMessageReceiverExceptionHandler.class);

--- a/src/main/docs/guide/pubsub/pullConsumer/errorHandling.adoc
+++ b/src/main/docs/guide/pubsub/pullConsumer/errorHandling.adoc
@@ -10,7 +10,7 @@ The framework provides a Global Error Handler api:gcp.pubsub.exception.DefaultPu
 There's two ways you can provide error handling.
 
 * Locally: your ann:gcp.pubsub.annotation.PubSubListener[] class implements api:gcp.pubsub.exception.PubSubMessageReceiverExceptionHandler[], then any error related to `Subscriptions` of that class will be handled by your class.
-* Globally: You define a new implementation of api:gcp.pubsub.exception.PubSubMessageReceiverExceptionHandler[] and annotate it as `@Primary` overriding the default one.
+* Globally: You define a new implementation of api:gcp.pubsub.exception.PubSubMessageReceiverExceptionHandler[] which link:https://docs.micronaut.io/latest/guide/#replaces[replaces] api:gcp.pubsub.exception.DefaultPubSubMessageReceiverExceptionHandler[].
 
 The api:exception.PubSubMessageReceiverException[] contains references to the originating bean that threw the exception as well as a reference to api:bind.PubSubConsumerState[] which has state information regarding the message handling.
 

